addons/product_images/wizard/product_fetch_image_wizard.py

```diff
@@ -0,0 +1,211 @@
+
+        default = super().default_get(fields_list)
+
+        ICP = self.env['ir.config_parameter']
+        google_pse_id = bool(ICP.get_param('google.pse.id'))
+        google_custom_search_key = bool(ICP.get_param('google.custom_search.key'))
+        if not google_pse_id or not google_custom_search_key:
+            raise UserError(_("The API Key and Search Engine ID must be set in the General "
+                              "Settings."))
+
+        if self._context.get('active_model') == 'product.template':
+            product_ids = self.env['product.template'].browse(
+                self._context.get('active_ids')).product_variant_ids
+        else:
+            product_ids = self.env['product.product'].browse(
+                self._context.get('active_ids'))
+        nb_product_selected = len(product_ids)
+        product_processable = product_ids.filtered(lambda p: not p.image_1920 and p.barcode)[:10000]
+        nb_product_processable = len(product_processable)
+        nb_product_unprocessable = nb_product_selected - nb_product_processable
+
+        default['product_processable'] = product_processable
+        # Not computed because the value of product_processable doesn't exist yet.
+        default['nb_product_selected'] = nb_product_selected
+        default['nb_product_processable'] = nb_product_processable
+        default['nb_product_unprocessable'] = nb_product_unprocessable
+
+        return default
+
+    product_processable = fields.Many2many(
+        comodel_name='product.product',
+        help="List of the selected product without a picture and with a barcode",
+    )
+    nb_product_selected = fields.Integer(string="Number of product selected", readonly=True)
+    nb_product_processable = fields.Integer(string="Number of product processable", readonly=True)
+    nb_product_unprocessable = fields.Integer(string="Number of product unprocessable",
+                                              readonly=True)
+
+    def action_fetch_image(self):
+        """ Trigger a cron job to fetch image on Google
+
+        The 10 first images are immediately fetched to improve the user experience. This way, they
+        can immediately view the images and be assured that the task is running well. Also, if any
+        error occurs, they can be thrown to the user. Then, a cron job is triggered to be run as
+        soon as possible, unless the daily request limit has been reached. In that case, the cron
+        job will be scheduled to run a day later.
+
+        :raises UserError: If the project is misconfigured on Google's side
+        :raises UserError: If the API Key or Search Engine ID is incorrect
+        :return: A notification to inform the user that the task is triggered
+        :rtype: `ir.actions.client`
+        """
+        self.product_processable.image_fetch_pending = True
+        try:
+            self._fetch_image_from_google(self.product_processable[:10])
+        except HTTPError as http_error:
```

Comment:
This error will be silently ignored if it's not handled by one of the `if`/`elif` below.